wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/pyramidannotation/timeslices/PyramidAnnotationTimeSliceController.java

```diff
@@ -0,0 +1,230 @@
+			@PageableDefault Pageable pageable,
+			PagedResourcesAssembler<PyramidAnnotationTimeSlice> assembler) {
+
+		Page<PyramidAnnotationTimeSlice> page = getPage(
+				pyramidAnnotationRepository.getTimeSlices(pyramidAnnotationId),
+				pageable);
+		for (PyramidAnnotationTimeSlice pats : page) {
+			processResource(pyramidAnnotationId, pats);
+		}
+		return new ResponseEntity<>(assembler.toResource(page), HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/{timeSliceId}", method = RequestMethod.GET)
+	public HttpEntity<PyramidAnnotationTimeSlice> getTimeSlice(
+			@PathVariable("pyramidAnnotationId") String pyramidAnnotationId,
+			@PathVariable("timeSliceId") int timeSliceId) {
+		PyramidAnnotationTimeSlice pats = pyramidAnnotationTimeSliceRepository
+				.findOne(pyramidAnnotationId, timeSliceId);
+		if (pats == null) {
+			throw new NotFoundException("Time slice " + timeSliceId
+					+ " not found in pyramid annotation " + pyramidAnnotationId);
+		}
+		processResource(pyramidAnnotationId, pats);
+		return new ResponseEntity<>(pats, HttpStatus.OK);
+	}
+
+	@RequestMapping(
+			value = "/{timeSliceId}/annotationPositions",
+			method = RequestMethod.GET)
+	public void getAnnotationPositions(
+			@PathVariable("pyramidAnnotationId") String pyramidAnnotationId,
+			@PathVariable("timeSliceId") int timeSliceId,
+			HttpServletResponse response) throws IOException {
+		File pyramidAnnotationFile = pyramidAnnotationTimeSliceRepository
+				.getAnnotationPositionsFile(pyramidAnnotationId, timeSliceId);
+		if (!pyramidAnnotationFile.exists()) {
+			response.sendError(404);
+			return;
+		}
+
+		response.setHeader("Content-disposition",
+				"attachment;filename=pyramid-annotation-" + pyramidAnnotationId +
+				"-" + pyramidAnnotationFile.getName());
+		response.setContentType("text/plain");
+		try (InputStream is = new BufferedInputStream(
+				new FileInputStream(pyramidAnnotationFile))) {
+			IOUtils.copy(is, response.getOutputStream());
+		}
+		response.flushBuffer();
+	}
+
+	@RequestMapping(
+			value = "/{timeSliceId}/annotationPositions",
+			method = RequestMethod.POST)
+	public PyramidAnnotation uploadAnnotationPositions(
+			@PathVariable("pyramidAnnotationId") String pyramidAnnotationId,
+			@PathVariable("timeSliceId") int timeSliceId,
+			@RequestParam("file") MultipartFile file) throws IOException {     
+		PyramidAnnotationTimeSlice timeSlice = new PyramidAnnotationTimeSlice(timeSliceId);
+		PyramidAnnotation pyramidAnnotation = pyramidAnnotationRepository.findById(pyramidAnnotationId).get();
+		File pyramidAnnotationFolder = new File(config.getPyramidAnnotationsFolder(), pyramidAnnotationId);
+
+		if(!pyramidAnnotationFolder.exists()) {
+			pyramidAnnotationFolder.mkdirs();
+		}
+
+		List<PyramidAnnotationTimeSlice> allTimeSlices = pyramidAnnotation.getTimeSlices();
+
+		if(containsTimeSlice(allTimeSlices, timeSliceId)) {
+			for (int i = 0; i < allTimeSlices.size(); i++) {
+				if(allTimeSlices.get(i).getSliceNumber() == timeSliceId) {
```

Comment:
The for loop is executed twice here.
Why not use the method `PyramidAnnotationTimeSlice findOne(String pyramidAnnotationId, int timeSlice)` from `PyramidAnnotationTimeSliceRepository` instead? It will return the time slice if found, null otherwise

Reply:
used findOne instead of custom method